.github/workflows/test-pull-request.yml

```diff
@@ -0,0 +1,57 @@
+# Run tests on a pull request after it has been labeled 'safe to test'
+
+name: test-pull-request
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on pull requests that contain the label 'safe to test'
+  pull_request_target:
+    types: [ labeled ]
+
+permissions: {}    
+
+jobs:
+  test-smoke:
+    name: smoke-${{ matrix.version }}-${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    if: contains(github.event.pull_request.labels.*.name, 'safe to test')
+    strategy:
+      fail-fast: false
+      matrix:
+        version: [R2021b, R2023a]
+        os: [windows-latest, ubuntu-latest]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+          ref: "refs/pull/${{ github.event.number }}/merge"
+          persist-credentials: false
+      - name: Verify Head
+        shell: bash
+        run: |
+          if [ "$(git ls-remote origin ${{ github.event.number }}/head | awk '{print $1}')" != "${{ github.event.pull_request.head.sha }}" ]; then
```

Comment:
Interesting. You might want to try confirming this works as expected by attempting to submit a change to a pull request right after you label it 'safe to test' but before the checkout action occurs.

Reply:
Thanks Mark! I forked my fork to test this and can confirm that the workflow now runs successfully on pull requests and fails in 'Verify Head' if a commit is made after the label is added but before the job runs 😄